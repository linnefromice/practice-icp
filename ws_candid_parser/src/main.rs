use std::path::Path;

use anyhow::Result;
use candid::pretty_check_file;

fn main() -> Result<()> {
    let candid_path = "assets/sample.did";
    let (env, actor) = pretty_check_file(Path::new(candid_path))?;
    let config = candid::bindings::rust::Config::new();

    let result = candid::bindings::rust::compile(&config, &env, &None);
    println!("{}", result);

    if let Some(actor) = actor {
        let result = candid::bindings::rust::compile(&config, &env, &Some(actor));
        println!("{}", result);
    }

    Ok(())
}
