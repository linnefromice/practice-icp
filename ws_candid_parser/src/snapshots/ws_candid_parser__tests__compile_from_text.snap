---
source: src/main.rs
expression: result
---
// This is an experimental feature to generate Rust binding from Candid.
// You may want to manually adjust some of the types.
#![allow(dead_code, unused_imports)]
use candid::{self, CandidType, Deserialize, Principal, Encode, Decode};
use ic_cdk::api::call::CallResult as Result;

candid::define_function!(pub FunctionType : (RequestArgsType) -> (
    ResponseType,
  ) query);
#[derive(CandidType, Deserialize)]
pub struct RequestArgsType(u64);

#[derive(CandidType, Deserialize)]
pub struct ResponseType { pub value: String, pub timestamp: u64 }


